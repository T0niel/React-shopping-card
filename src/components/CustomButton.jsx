export default function CustomButton({onClick, text}){
    return (
      <button
        className="block w-[100%] border-2 opacity-90 hover:opacity-100 border-gray-400 rounded-lg p-2"
        onClick={onClick}
      >
        {text}
      </button>
    );
}