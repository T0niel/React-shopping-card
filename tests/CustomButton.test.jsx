import { describe, expect, it, vi } from 'vitest';
import CustomButton from '../src/components/CustomButton';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('CustomButton', () => {
    it('Matches snapshot', () => {
        const { container } = render(
          <CustomButton onClick={() => {}} text="Hello" />
        );

        expect(container).toMatchSnapshot();
    })

    it('Renders the text', () => {
        render(<CustomButton onClick={() => {}} text='Hello'/>);

        const button = screen.queryByText('Hello');
        expect(button).not.toEqual(null);
    })

    it('Calls onClick when button clicked', async () => {
        const user = userEvent.setup();
        const onClick = vi.fn();
        render(<CustomButton onClick={onClick} text="Hello" />);
        
        const button = screen.queryByText('Hello');
        await user.click(button);

        expect(onClick.mock.calls.length).toEqual(1)
    })
    
    it('it does not call onClick when button not clicked', () => {
      const onClick = vi.fn();
      render(<CustomButton onClick={onClick} text="Hello" />);

      expect(onClick.mock.calls.length).toEqual(0);
    });
})