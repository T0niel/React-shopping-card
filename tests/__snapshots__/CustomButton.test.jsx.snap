// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomButton > Matches snapshot 1`] = `
<div>
  <button
    class="block w-[100%] border-2 opacity-90 hover:opacity-100 border-gray-400 rounded-lg p-2"
  >
    Hello
  </button>
</div>
`;
